qiskit/transpiler/preset_passmanagers/generate_preset_pass_manager.py

```diff
@@ -460,7 +470,15 @@ def _parse_basis_gates(basis_gates, backend, inst_map, skip_target):
             return None, name_mapping, skip_target
 
         for inst in instructions:
-            if inst not in standard_gates or inst not in default_gates:
+            if inst not in standard_gates and inst not in default_gates:
```

Comment:
After this change I noticed that there are 2-5 flaky unit tests that panic when running unitary synthesis. All tests were passing before changing this `or` to an `and`, but I believe this was an oversight from previous iterations of the code that was causing the target to be skipped most of the time basis gates were manually provided. Now the target is not skipped, which means the rust variant of unitary synthesis is run, and I am afraid this could have uncovered a bug in unitary synthesis. I will take a closer look.